js/src/edit.js

```diff
@@ -62,22 +65,77 @@ class Edit {
 		);
 	}
 
+	/**
+	 * Initialize the store and filters.
+	 *
+	 * @returns {void}
+	 * @private
+	 */
 	_init() {
 		this._store = this._registerStoreInGutenberg();
 
 		this._registerCategorySelectorFilter();
-
 		this._registerPlugin();
 
 		this._data = this._initializeData();
 
-		this._store.dispatch( setSettings( {
+		this._store.dispatch( actions.setSettings( {
 			snippetEditor: {
 				baseUrl: this._args.snippetEditorBaseUrl,
+				permalink: this._args.snippetEditorPermalink,
 				date: this._args.snippetEditorDate,
 				recommendedReplacementVariables: this._args.recommendedReplaceVars,
 			},
 		} ) );
+
+		if ( window.wpseoPrimaryCategoryL10n ) {
+			this._store.dispatch( actions.updateData( {
+				parents: this._getPrimaryCategoryParents(),
+				primaryTaxonomySlug: this._getPrimaryCategorySlug(),
+			} ) );
+		}
+	}
+
+	/**
+	 * Gets the primary category data from the localized dataset.
+	 *
+	 * @returns {Object} The category data.
+	 * @private
+	 */
+	_getPrimaryCategory() {
+		return wpseoPrimaryCategoryL10n.taxonomies.category;
+	}
+
+	/**
+	 * Gets the primary category slug.
+	 *
+	 * @returns {string} The slug.
+	 * @private
+	 */
+	_getPrimaryCategorySlug() {
+		const categoryData = this._getPrimaryCategory();
+
+		if ( ! categoryData ) {
+			return "";
+		}
+
+		return Object.values( categoryData.terms ).find( term => term.id === categoryData.primary ).slug;
+	}
+
+	/**
+	 * Gets the primary category's parents.
+	 *
+	 * @returns {number} The parent. 0 if no parent is set.
+	 * @private
+	 */
+	_getPrimaryCategoryParents() {
+		const categoryData = this._getPrimaryCategory();
+
+		if ( ! categoryData ) {
+			return 0;
+		}
+
+		return Object.values( categoryData.terms ).find( term => term.id === categoryData.primary ).parents;
```

Comment:
I see this piece `Object.values( categoryData.terms ).find( term => term.id === categoryData.primary )` twice. I would choose to make a function for it. I also think this line is pretty hard to read.

Another concern I have is what if `categoryData.primary` isn't in the `categoryData.terms`?

Reply:
`primary` is always set to a (sane) default before localization.